wallet-next/src/storage.rs

```diff
@@ -34,15 +38,69 @@ VALUES ( ?1 )
     pub async fn read_table(self: &Storage) -> anyhow::Result<String> {
         let recs = sqlx::query!(
             r#"
-SELECT id, value
-FROM penumbra
-ORDER BY id
-LIMIT 1
+            SELECT id, value
+            FROM penumbra
+            ORDER BY id
+            LIMIT 1
         "#
         )
         .fetch_all(&self.pool)
         .await?;
 
         Ok(recs[0].value.clone())
     }
+    /// The last block height we've scanned to, if any.
+    pub async fn last_sync_height(&self) -> anyhow::Result<Option<u64>> {
+        let result = sqlx::query!(
+            r#"
+            SELECT height
+            FROM sync_height
+            ORDER BY height DESC
+            LIMIT 1
+        "#
+        )
+        .fetch_all(&self.pool)
+        .await?;
+
+        Ok(result[0].value.clone())
+    }
+    pub async fn chain_params(&self) -> anyhow::Result<Vec<u8>> {
+        let result = query!(
+            r#"
+            SELECT bytes
+            FROM chain_params
+            LIMIT 1
+        "#
+        )
+        .fetch_all(&self.pool)
+        .await?;
+
+        Ok(result[0].value.clone())
+    }
+    pub async fn full_viewing_key(&self) -> anyhow::Result<Vec<u8>> {
+        let result = query!(
+            r#"
+            SELECT bytes
+            FROM full_viewing_key
+            LIMIT 1
+            "#
+        )
+        .fetch_all(&self.pool)
+        .await?;
+
+        Ok(result)
+    }
+    pub async fn note_commitment_tree(&self) -> anyhow::Result<Vec<u8>> {
+        let result = query!(
+            r#"
+            SELECT bytes
+            FROM note_commitment_tree
+            LIMIT 1
+            "#
+        )
+        .fetch_all(&self.pool)
+        .await?;
+
+        Ok(result)
```

Comment:
This should return a `NoteCommitmentTree`; unlike the other methods, the `incrementalmerkletree`-provided `NoteCommitmentTree` isn't proto-serialized, but it can be serialized with `bincode` similarly to the existing `get_nct` and `put_nct` methods on the shielded pool component by doing
```
let nct_data = bincode::serialize(&self.note_commitment_tree)?;
```
on the way in and
```
bincode::deserialize(&bytes).map_err(Into::into)
```
on the way out.

This is awkward, but any serialization code for the current NCT will be rewritten soon when we implement the NCT using the TCT library, so that serialization code has a lifespan of at most several weeks.